src/tree_data.c

```diff
@@ -2779,6 +2779,83 @@ lyd_target(const struct ly_path *path, const struct lyd_node *tree)
     return (struct lyd_node_term *)target;
 }
 
+/**
+ * @brief Check the equality of the two schemas from different contexts.
+ *
+ * @param schema1 of first node.
+ * @param schema2 of second node.
+ * @return 1 if the schemas are equal otherwise 0.
+ */
+static ly_bool
+lyd_compare_schema_equal(const struct lysc_node *schema1, const struct lysc_node *schema2)
+{
+    if (!schema1 && !schema2) {
+        return 1;
+    } else if (!schema1 || !schema2) {
+        return 0;
+    }
+
+    assert(schema1->module->ctx != schema2->module->ctx);
+
+    if (schema1->nodetype != schema2->nodetype) {
+        return 0;
+    }
+
+    if (schema1->module->latest_revision != schema2->module->latest_revision) {
```

Comment:
Ah, another thing I forgot about before, but no harm done in this case. This is redundant and actually wrong, ignore those `latest_revision` flags, they have no relevance for this purpose.

Reply:
It is interesting that it has to be deleted. If I delete it, it does not pass tests in sysrepo. Are you sure about that?

Reply:
Yes, pretty sure. Please, never adjust the code for the tests without understanding the specific test-case. So, when you understand what exactly is going on (what I can help with), let me know and we can discuss what to do.